tests/unit_tests/explore/api_test.py

```diff
@@ -0,0 +1,30 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any
+
+import pytest
+
+
+def test_explore_datasource_not_found(client: Any, full_api_access: None) -> None:
+    # validating the payload for a dataset that doesn't exist
+    # user should be expecting missing_datasource view
+    response = client.get(
+        "/api/v1/explore/?dataset_id=50000&dataset_type=table",
+    )
+    response.json["result"]["dataset"]["name"] == "[Missing Dataset]"
+    assert response.status_code == 200
```

Comment:
Does this mean we are expecting a `200` when no dataset was found?

Reply:
Yeah same question.

Reply:
yea this is the expected results whenever we hit explore with a unknown dataset, assuming product is okay with this for now but if this needs to change atleast we have a test to warrant expected result